core/src/mindustry/logic/LExecutor.java

```diff
@@ -557,6 +571,13 @@ public void run(LExecutor exec){
 
             if(from instanceof MemoryBuild mem && (exec.privileged || (from.team == exec.team && !mem.block.privileged))){
                 output.setnum(address < 0 || address >= mem.memory.length ? 0 : mem.memory[address]);
+            }else if(from instanceof LogicBuild logic && (exec.privileged || from.team == exec.team) && position.isobj && position.objval instanceof String name){
+                LVar fromVar = logic.executor.optionalVar(name);
+                if(fromVar != null && !output.constant){
+                    output.objval = fromVar.objval;
```

Comment:
this doesn't check if the executor you're reading is itself privileged, which means that same-team world processors may be readable, potentially leaking secret variables to an unprivileged context

Reply:
> this doesn't check if the executor you're reading is itself privileged

I was aware when I merged this, which is why it was fixed as the next commit.